client/src/tpu_client.rs

```diff
@@ -126,29 +112,24 @@ impl TpuClient {
         config: TpuClientConfig,
         connection_cache: Arc<ConnectionCache>,
     ) -> Result<Self> {
-        let nonblocking_rpc_client = rpc_client.get_nonblocking_client_copy();
-
         let create_tpu_client = NonblockingTpuClient::new_with_connection_cache(
-            nonblocking_rpc_client,
+            rpc_client.rpc_client.clone(),
             websocket_url,
             config,
             connection_cache,
         );
-
-        let runtime = tokio::runtime::Builder::new_current_thread()
-            .thread_name("tpu-client")
-            .enable_io()
-            .enable_time()
-            .build()
-            .unwrap();
-        let _guard = runtime.enter();
-        let tpu_client = runtime.block_on(create_tpu_client)?;
+        let tpu_client = tokio::task::block_in_place(|| {
+            rpc_client
+                .runtime
```

Comment:
Please use a getter that returns a ref

Reply:
Is there a particular reason that you're recommending this? The client's runtime is already accessed through an `Arc`.  Do you mean to do something like:

```
pub(crate) fn get_runtime(&self) -> &Runtime {
    self.runtime.as_ref().expect("runtime")
}
```

To reduce code repetition?

Reply:
Yes, that code looks good. Yeah, it's to reduce code repetition, but also just as good practice IHMO, and to keep it clear that the rpc_client owns the runtime, and that the runtime is not shared like it would be with an Arc.

Reply:
Ok great, I ended up changing both accesses to be clearer, let me know what you think